components/tyk-bootstrap/templates/bootstrap-serviceaccount.yml

```diff
@@ -1,8 +1,8 @@
-{{- if .Values.bootstrap -}}
+{{- if and .Values.bootstrap .Values.global.serviceAccountName -}}
```

Comment:
As far as i know, the job needs this serviceaccount. So, we do not need to check if `serviceAccountName` exists or not. It should only affect metadata for this Job

Reply:
if the intention is to make it configurable, it doesn't hurt is we have to check if certain things exists in order to ensure that certain things are not left out in the process of the configuring it.

Reply:
will this approach be backward compatible? Because old users will not have `serviceAccountName` defined in their `values.yaml` file; so, this `if` check will fail. I know that upgrades won't work with chart hooks but essentially I see potential breaking change here

Reply:
Ha I see. This won't work if it has to be backward compatible.